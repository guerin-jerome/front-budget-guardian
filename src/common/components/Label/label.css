label {
  font-family: Arial, sans-serif;
  font-size: 16px;
  font-weight: bold;
  color: #333;
}
