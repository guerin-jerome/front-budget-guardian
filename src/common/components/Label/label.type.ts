import { PropsWithChildren } from "react";

export type LabelProps = PropsWithChildren & {
  htmlFor: string;
};
