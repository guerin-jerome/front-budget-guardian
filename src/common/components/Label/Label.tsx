import { LabelProps } from "./label.type";
import "./label.css";

export const Label = ({ htmlFor, children }: LabelProps) => (
  <label htmlFor={htmlFor}>{children}</label>
);
