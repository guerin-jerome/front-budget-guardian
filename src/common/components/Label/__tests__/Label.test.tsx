import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { Label } from "../Label";

describe("<Label />", () => {
  it("matchs snapshot", () => {
    const { container } = render(<Label htmlFor="test">Hello</Label>);
    expect(container).toMatchSnapshot();
  });

  it("renders well", () => {
    render(<Label htmlFor="test">Hello</Label>);
    const labelText = screen.getByText("Hello");
    expect(labelText).toBeDefined();
  });
});
