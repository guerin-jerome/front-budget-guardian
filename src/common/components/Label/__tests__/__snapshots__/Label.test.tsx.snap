// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Label /> > matchs snapshot 1`] = `
<div>
  <label
    for="test"
  >
    Hello
  </label>
</div>
`;
